hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/utils/TestHeader.java

```diff
@@ -0,0 +1,123 @@
+package org.apache.hadoop.fs.azurebfs.utils;
+
+import org.apache.hadoop.fs.azurebfs.constants.AbfsOperations;
+import org.apache.hadoop.fs.azurebfs.constants.FileSystemConfigurations;
+import org.assertj.core.api.Assertions;
+
+public class TestHeader implements Listener {
+  String prevContinuationHeader = "";
+  int prevRetryCount = 0;
+  String clientCorrelationID;
+  String fileSystemID;
+  String streamID = "";
+  String operation;
+  int maxRetryCount = FileSystemConfigurations.DEFAULT_MAX_RETRY_ATTEMPTS;
+  String GUID_PATTERN = "[0-9a-fA-F]{8}-([0-9a-fA-F]{4}-){3}[0-9a-fA-F]{12}";
+  // client-req-id as per docs: ^[{(]?[0-9a-f]{8}[-]?([0-9a-f]{4}[-]?)
+  // {3}[0-9a-f]{12}[)}]?$
+
+  @Override
+  public void afterOp(String header) {
+    System.out.println("testing header: " + header);
+    testBasicFormat(header);
+
+    if (isContinuationOp(header)) {
```

Comment:
testContinuationHeaderConsistency as a the only top level function. Within that first call testBasicFormat and then do special checking based on continuationOp etc